spec/common/testapp_module.js

```diff
@@ -0,0 +1,79 @@
+/* global console, jasmine, define, beforeAll, afterAll, global, __ABLY__, Ably */
+"use strict";
+
+/* Shared test helper used within Jasmine tests */
+
+define(['testapp'], function(testAppModule) {
+  var isBrowser = (typeof(window) === 'object');
+
+  function globalObject() {
+    if (isBrowser) {
+      return window;
+    } else {
+      return global;
+    }
+  }
+
+  function updateTestApp(newTestApp) {
+    globalObject().AblyTestApp = newTestApp;
+  }
+
+  function getTestApp() {
+    return globalObject().AblyTestApp;
+  }
+
+  function incrementBeforeAllCounter(quantity) {
+    if (!globalObject().AblyBeforeAllCounter) { globalObject().AblyBeforeAllCounter = 0; }
```

Comment:
You could move this out of this function right next to the definition of `isBrowser` to initialise this variable.